lib/systemservice/systemd.go

```diff
@@ -244,10 +244,23 @@ func (s *systemdManager) IsPackageServiceInstalled(pkg loc.Locator) (bool, error
 }
 
 // ListPackageServices lists installed package services
-func (s *systemdManager) ListPackageServices() ([]PackageServiceStatus, error) {
+func (s *systemdManager) ListPackageServices(opts ListServiceOptions) ([]PackageServiceStatus, error) {
 	var services []PackageServiceStatus
 
-	out, err := invokeSystemctl("list-units", "--plain", "--no-legend")
+	args := []string{"list-units", "--plain", "--no-legend"}
+	if opts.All {
+		args = append(args, "--all")
+	}
+	if opts.Type != "" {
+		args = append(args, "--type", opts.Type)
+	}
+	if opts.State != "" {
+		args = append(args, "--state", opts.State)
+	}
+	if opts.Pattern != "" {
+		args = append(args, opts.Pattern)
```

Comment:
Should this be quoted maybe?

Reply:
I don't believe so - system `exec` will be taking care of that.